storage/src/test/java/org/apache/kafka/tiered/storage/utils/BrokerLocalStorage.java

```diff
@@ -141,7 +146,11 @@ private boolean isOffsetPresentInFirstLocalSegment(TopicPartition topicPartition
         if (offsetToSearch.equals(firstLogFileBaseOffset)) {
             return true;
         }
-        File partitionDir = new File(brokerStorageDirectory.getAbsolutePath(), topicPartition.toString());
+        File partitionDir = brokerStorageDirectorys.stream()
+                .filter(dir -> dirContainsTopicPartition(topicPartition, dir))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(String.format("[BrokerId=%d] Directory for the topic-partition %s " +
+                "was not found", brokerId, topicPartition)));
```

Comment:
previously, we were returning the `partitionDir` instead of  `logDir`:

```suggestion
File logDir = brokerStorageDirectorys.stream()
                .filter(dir -> dirContainsTopicPartition(topicPartition, dir))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException(String.format("[BrokerId=%d] Directory for the topic-partition %s " +
                "was not found", brokerId, topicPartition)));
File partitionDir = new File(logDir.getAbsolutePath(), topicPartition.toString());
```